src/schemas/clinical-resolvers/clinicalData.ts

```diff
@@ -0,0 +1,80 @@
+import { getPaginatedClinicalData, ClinicalDataVariables } from '../../clinical/clinical-service';
+import { ClinicalEntityData, ClinicalInfo } from '../../clinical/clinical-entities';
+import { ClinicalErrorsResponseRecord } from '../../common-model/entities';
+import { errorResolver } from './clinicalErrors';
+
+export type ClinicalEntityGQLData = {
+  programShortName: string;
+  clinicalEntities: ClinicalEntityDisplayData[];
+};
+
+// FE Clinical Data Query Response Payload
+export type ClinicalEntityDataResponse = ClinicalEntityGQLData & {
+  clinicalErrors?: ClinicalErrorsResponseRecord[];
+};
+
+// GQL Formatting
+type EntityDisplayRecord = { name: string; value: string };
+
+interface ClinicalEntityDisplayData extends Omit<ClinicalEntityData, 'records'> {
+  records: EntityDisplayRecord[][];
+}
+
+const convertClinicalDataToGql = (
+  programShortName: string,
+  clinicalEntities: ClinicalEntityData[],
+) => {
+  const clinicalDisplayData: ClinicalEntityDisplayData[] = clinicalEntities.map(
+    (entity: ClinicalEntityData) => {
+      const records: EntityDisplayRecord[][] = [];
```

Comment:
what is the double `[][]` my google skills are failing me?

Reply:
An array containing arrays containing EntityDisplayRecords
`[ [ {} ], [ {} ], [ {} ] ]`